internal/topo/node/source_pool.go

```diff
@@ -296,6 +314,20 @@ func (ss *sourceSingleton) attach(instanceKey string, bl int) error {
 	return err
 }
 
+func (ss *sourceSingleton) detachSchema(schemaKey string) {
+	ss.Lock()
+	defer ss.Unlock()
+	decodeConverter := ss.ctx.Value(kctx.DecodeKey)
+	if decodeConverter != nil {
+		fastDecoder, ok := decodeConverter.(*json.FastJsonConverter)
```

Comment:
Should not refer to specific converter. Add AttachSchema and DetachSchema to the converter interface?